cdc/entry/mounter.go

```diff
@@ -219,29 +219,38 @@ func (m *mounterImpl) unmarshalRowKVEntry(tableInfo *model.TableInfo, rawKey []b
 	}, nil
 }
 
-const (
-	ddlJobListKey         = "DDLJobList"
-	ddlAddIndexJobListKey = "DDLJobAddIdxList"
-)
-
-// UnmarshalDDL unmarshals the ddl job from RawKVEntry
-func UnmarshalDDL(raw *model.RawKVEntry) (*timodel.Job, error) {
-	if raw.OpType != model.OpTypePut || !bytes.HasPrefix(raw.Key, metaPrefix) {
+// ParseJob parses the job from the raw KV entry.
+func ParseJob(tblInfo *model.TableInfo, rawKV *model.RawKVEntry, id int64) (*timodel.Job, error) {
+	if rawKV.OpType != model.OpTypePut {
 		return nil, nil
 	}
-	meta, err := decodeMetaKey(raw.Key)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-	if meta.getType() != ListData {
-		return nil, nil
-	}
-	k := meta.(metaListData)
-	if k.key != ddlJobListKey && k.key != ddlAddIndexJobListKey {
-		return nil, nil
+
+	var v []byte
+	if bytes.HasPrefix(rawKV.Key, metaPrefix) {
+		// old queue base job.
+		v = rawKV.Value
+	} else {
+		recordID, err := tablecodec.DecodeRowKey(rawKV.Key)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		if len(rawKV.Value) == 0 {
```

Comment:
When can `len(rawKV.Value) == 0` happen?

Reply:
no, it should never happen, I will remove it